swap/peer.go

```diff
@@ -76,25 +77,7 @@ func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg *EmitChequeMsg) err
 	log.Info("received emit cheque message")
 
 	cheque := msg.Cheque
-	if cheque.Contract != sp.contractAddress {
-		return fmt.Errorf("wrong cheque parameters: expected contract: %s, was: %s", sp.contractAddress, cheque.Contract)
-	}
-
-	// the beneficiary is the owner of the counterparty swap contract
-	if err := sp.swap.verifyChequeSig(cheque, sp.beneficiary); err != nil {
-		log.Error("error invalid cheque", "from", sp.ID().String(), "err", err.Error())
-		return err
-	}
-
-	if cheque.Beneficiary != sp.swap.owner.address {
-		return fmt.Errorf("wrong cheque parameters: expected beneficiary: %s, was: %s", sp.swap.owner.address, cheque.Beneficiary)
-	}
-
-	if cheque.Timeout != 0 {
-		return fmt.Errorf("wrong cheque parameters: expected timeout to be 0, was: %d", cheque.Timeout)
-	}
-
-	// TODO: check serial and balance are higher
+	sp.processAndVerifyCheque(cheque)
```

Comment:
`processAndVerifyCheque` returns an error but this is not checked here

Reply:
You're right, must have forgot to put the check when I extracted the function